packages/client/src/subscribe/index.ts

```diff
@@ -646,11 +646,15 @@ export class Subscriber {
         const realtimeMessageStream = messageStream(this.client.connection, options)
 
         // cancel both streams on end
-        async function end(optionalErr: Todo) {
-            await Promise.all([
+        async function end(optionalErr?: Error) {
+            const tasks = [
                 resendMessageStream.cancel(optionalErr),
                 realtimeMessageStream.cancel(optionalErr),
-            ])
+                resendSubscribeSub.cancel(optionalErr)
+            ]
+
+            await Promise.allSettled(tasks)
+            await Promise.all(tasks)
```

Comment:
Not sure I understand this pattern of doing first `allSettled` followed by `all`.

Reply:
https://twitter.com/timkevinoxley/status/1389940552209256448